### PR TITLE
Cosmos - Nested PartitionKey Support

### PR DESCRIPTION
**Pull request by ksivamuthu**
# Description

Read the property value from the request data using jsonpath to support nested partition keys.

Format for configuring nested partition key is

"address.zip" - fields separated by dot - similar to a .NET Bulk executor library

Example: 

Configuration: 
```
apiVersion: dapr.io/v1alpha1
kind: Component
metadata:
  name: send-event
spec:
  type: bindings.azure.cosmosdb
  metadata:
  - name: url 
    value: URL
  - name: masterKey    # Azure Storage Account Name
    value: MASTER_KEY
  - name: database     # Azure Storage Account Key
    value: DB
  - name: collection  # Azure Storage Container Name
    value: CONTAINER
  - name: partitionKey
    value: "address.zip"
```

JSON: 
```
{
    "data": {
    	"id": "1",
        "name": "test",
        "address": {
             "zip": "35220"
        }
    }
}
```
image

## Issue reference

PR will close: #288 

## Checklist

Please make sure you've completed the relevant tasks for this PR, out of the following list:

* [x] Code compiles correctly
* [x] Created/updated tests
* [ ] Extended the documentation

### CODE REVIEW

File: bindings/azure/cosmosdb/cosmosdb.go [hunk 1]
```diff
@@ -12,6 +12,7 @@ import (
 	"github.com/a8m/documentdb"
 	"github.com/dapr/components-contrib/bindings"
 	"github.com/dapr/dapr/pkg/logger"
+	"github.com/yalp/jsonpath"
```
Comment (yaron2) [inline bindings/azure/cosmosdb/cosmosdb.go:15]: Is there a way to do this without using this specific package? it looks unmaintained.

Comment (yaron2) [reply bindings/azure/cosmosdb/cosmosdb.go:15]: In general, we want as little external dependencies if possible..

Comment (ksivamuthu) [reply bindings/azure/cosmosdb/cosmosdb.go:15]: That makes sense. I will programmatically parse the fields and navigate to the nested map.

Comment (ksivamuthu) [reply bindings/azure/cosmosdb/cosmosdb.go]: Done the changes. Please review.

**Comment by yaron2** [comment]
Thanks!

**Pull request merged**